lib/reviewlette.rb

```diff
@@ -1,202 +1,53 @@
-    end
-
-    # Sets instance variables.
-    def setup
-      @logger = Logger.new('review.log')
-      @db = Reviewlette::Database.new
-      @github_connection = Reviewlette::GithubConnection.new
-      @trello_connection = Reviewlette::TrelloConnection.new
-      @board = Trello::Board.find(TRELLO_CONFIG1['board_id'])
-    end
-
-    # Finds a sets card.
-    def find_id
-      if @id != 0
-        @card = @trello_connection.find_card_by_id(@id)
-        true
       else
-        @logger.warn("Id not found, skipping Issue #{@title} with number #{@number}")
-        false
-      end
-    end
-
-    # Selects and sets reviewer.
-    def set_reviewer
-      begin
-        while !(@reviewer)
-          @reviewer = @trello_connection.determine_reviewer(@card) if @card
-        end
-        @trelloname = @reviewer.username
-        puts "Selected #{@reviewer.username}"
-        return true
-      rescue AlreadyAssignedException => e
-        @logger.warn("Skipped Issue #{@card.short_id} because #{e.message}")
-        puts ("Skipped Issue #{@card.short_id} because #{e.message}")
-        return false
+        puts "No matching card found (id #{card_id})"
       end
     end
 
-    # Get Trelloname from configfile.
-    def transform_name
-      @githubname = @db.find_gh_name_by_trello_name(@trelloname)
-    end
-
-    # Adds Assignee on GitHub.
-    def add_reviewer_on_github
-      @github_connection.add_assignee(@repo, @number, @title, @body, @githubname) if @number && @githubname
-    end
-
-    # Comments on GitHub.
-    def comment_on_github
-      @github_connection.comment_on_issue(@repo, @number, @githubname, @card.url) if @number && @githubname
-    end
-    # Adds Reviewer on Trello Card.
-    def add_to_trello_card
-      @trello_connection.add_reviewer_to_card(@reviewer, @card)
-    end
-
-    # Comments on Trello Card.
-    def comment_on_trello
-      @full_comment = '@' + @trelloname + ' will review ' + 'https://github.com/'+ @repo+'/issues/'+@number.to_s
-      @trello_connection.comment_on_card(@full_comment, @card) if @full_comment
-    end
+  end
 
-    # TODO: More generic 'Done' and 'in Review' are not present everywhere
-    def move_to_list
-      if @github_connection.pull_merged?(@repo, @id)
-        @column = @trello_connection.find_column('Done')
-        @trello_connection.move_card_to_list(@card, @column)
-      else
-        @column = @trello_connection.find_column('In review')
-        @trello_connection.move_card_to_list(@card, @column)
-      end
-    end
+  def select_reviewer(issue, card)
+    reviewers = MEMBERS_CONFIG['members']
+    # remove people on vacation
+    members_on_vacation = Vacations.members_on_vacation
+    reviewers = reviewers.reject{|r| members_on_vacation.include? r['suse_username'] }
+    # remove trello card owner
+    reviewers = reviewers.reject{|r| card.members.map(&:username).include? r['trello_username'] }
+    reviewer = reviewers.sample
```

Comment:
@digitaltom I don't think having the sample here is better as it was before. We have to count a number of reviews and based on reviewed_at date select one from the list of available members